src/codewhisperer/util/telemetryHelper.ts

```diff
@@ -137,4 +148,84 @@ export class TelemetryHelper {
     public isTelemetryEnabled(): boolean {
         return globals.telemetry.telemetryEnabled
     }
+
+    public resetClientComponentLatencyTime() {
+        this.invokeSuggestionStartTime = 0
+        this.sdkApiCallStartTime = 0
+        this.sdkApiCallEndTime = 0
+        this.fetchCredentialStartTime = 0
+        this.firstSuggestionShowTime = 0
+        this.allPaginationEndTime = 0
+        this.firstResponseRequestId = ''
+        this.sessionId = ''
+    }
+
+    public setInvokeSuggestionStartTime() {
+        if (this.invokeSuggestionStartTime === 0) {
+            this.invokeSuggestionStartTime = performance.now()
+        }
+    }
```

Comment:
Setting values once can cause data quality issues if the state isn't reset after a bad execution. For example, what if `getPaginatedRecommendation` failed for whatever reason? The next successful execution would use an inaccurate timestamp.

If we don't need to handle concurrency then it's safe to always reset state at the start of the execution:
```suggestion
    /** This method is assumed to be invoked first at the start of execution **/
    public setInvokeSuggestionStartTime() {
         this.resetClientComponentLatencyTime()
         this.invokeSuggestionStartTime = performance.now()
    }
```